src/pages/workspaces/workspace/workspace-utils.ts

```diff
@@ -0,0 +1,36 @@
+
```

Comment:
Since workspace is passed around and used by so many functions and components, I still think these types should live at a higher level than pages/workspaces/workspace.

Reply:
Hmm.. I thought you were talking about the cloud provider stuff originally but I see your point. I moved it to the same level the existing `workspace-utils.js` is. I made a file `workspace-utils.d.ts`  for this, is that the proper usage of a `d.ts` extension? It seems to work as intended and allow for slow migration from the `.js` definitions to `.ts`...

Reply:
Let's make it a plain `.ts` TypeScript file. [Declaration files](https://www.typescriptlang.org/docs/handbook/declaration-files/introduction.html) (`.d.ts` files) allow providing TypeScript types for code written in JS. Generally, `module.d.ts` would be paired with `module.js`. In this case, there's no `libs/workspace-utils.js` file (though there is a `components/workspace-utils.js`).